client_metrics.go

```diff
@@ -127,50 +131,54 @@ func (m *ClientMetrics) Collect(ch chan<- prom.Metric) {
 // EnableClientHandlingTimeHistogram turns on recording of handling time of RPCs.
 // Histogram metrics can be very expensive for Prometheus to retain and query.
 func (m *ClientMetrics) EnableClientHandlingTimeHistogram(opts ...HistogramOption) {
-	for _, o := range opts {
-		o(&m.clientHandledHistogramOpts)
-	}
-	if !m.clientHandledHistogramEnabled {
-		m.clientHandledHistogram = prom.NewHistogramVec(
-			m.clientHandledHistogramOpts,
-			[]string{"grpc_type", "grpc_service", "grpc_method"},
-		)
-	}
-	m.clientHandledHistogramEnabled = true
+	m.doOnceClientHandledHistogramEnable.Do(func() {
+		for _, o := range opts {
+			o(&m.clientHandledHistogramOpts)
```

Comment:
I wasn't sure if the `opts` handling should be within this block, but I figured there's no point in processing them a second time since the subsequent code will bail and never read them.